services/wallet/bridge/cbridge.go

```diff
@@ -251,7 +256,14 @@ func (s *CBridge) EstimateGas(fromNetwork *params.Network, toNetwork *params.Net
 
 	estimation, err := ethClient.EstimateGas(ctx, msg)
 	if err != nil {
-		return 0, err
+		if !token.IsNative() {
+			// TODO: this is a temporary solution until we find a better way to estimate the gas
+			// hardcoding the estimation for other than ETH, cause we cannot get a proper estimation without having an approval placed first
+			// this is an error we're facing otherwise: `execution reverted: ERC20: transfer amount exceeds allowance`
+			estimation = 350000
```

Comment:
do we need this here?

also is this value accurate for cbridge?

Reply:
> do we need this here?

Yes, we do, cause in case of erc20 tokens we will get `execution reverted: ERC20: transfer amount exceeds allowance` error, this way we use hardcoded value and don't skip this path in the Router logic.


We need it, but the value is not accurate, used the same one as we used in the Hop.

In both, Hop and Celar, would be good to have logic for returning "hardcoded" gas estimated values, like check how much is the gas for ETH and some erc20 token for L1->L2, L2-L2, L2-L1 and based on that create a map with hardcoded values and increase them a bit and use that values as estimated gas (to be closer to a real estimation). That's something we will do later.

But for now, I just left it to this value.